plugin/ranger.vim

```diff
@@ -23,36 +23,42 @@
 
 
 " ================ Ranger =======================
+if !exists('g:ranger_path')
+    let g:ranger_path = 'ranger'
+endif
+
 if has('nvim')
-  function! OpenRanger()
-    let currentPath = expand("%:p:h")
-    let rangerCallback = { 'name': 'ranger' }
-    function! rangerCallback.on_exit(id, code)
-      Bclose!
-      try
-        if filereadable('/tmp/chosenfile')
-          exec system('sed -ie "s/ /\\\ /g" /tmp/chosenfile')
-          exec 'argadd ' . system('cat /tmp/chosenfile | tr "\\n" " "')
-          exec 'edit ' . system('head -n1 /tmp/chosenfile')
-          call system('rm /tmp/chosenfile')
-        endif
-      endtry
+    function! OpenRanger(dir)
+        let currentPath = expand(a:dir)
+        let tmp_file_path = tempname()
+        let rangerCallback = { 'name': 'ranger' , 'tmp_file_path': tmp_file_path}
+        function! rangerCallback.on_exit(id, code)
+            bdelete!
```

Comment:
We should keep Bclose! here because the behaviour is not the same. If I split a tab in two windows and then I open un file in one window with ranger, the window in delete and the file is open in the other window... Can you try this? Do you use neovim?